fluentd/configs.d/filter/k8s_record_transform.conf

```diff
@@ -3,7 +3,7 @@
   enable_ruby
   <record>
     hostname ${(kubernetes_host rescue nil) || File.open('/etc/docker-hostname') { |f| f.readline }.rstrip}
-    message ${log}
+    message ${(message rescue nil) || log}
     version 1.1.4
   </record>
   remove_keys log,stream
```

Comment:
probably fine, but note - we're deleting "log" whether or not it's been preserved. if the metadata has both "message" and "log" we throw the second away.

i can't think of a better approach though other than maybe concatenating them.